src/Entities/Pawn.cpp

```diff
@@ -106,11 +106,54 @@ void cPawn::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 
 
+bool cPawn::DeductTotem(cPlayer * a_Player)
+{
+	auto & inv = a_Player->GetInventory();
+	if (inv.GetEquippedItem().m_ItemType == E_ITEM_TOTEM_OF_UNDYING)
+	{
+		inv.SetEquippedItem(cItem());
+		return true;
+	}
+	if (inv.GetShieldSlot().m_ItemType == E_ITEM_TOTEM_OF_UNDYING)
```

Comment:
Wait, this means that if you hold a totem of undying in both your main hand _and_ the offhand, both will be used right?
I just tested it in vanilla and this is not correct behaviour, only one should be used (the one in the main hand).

So my suggestion would be making this an else if.